src/components/follow-list/AllCheckBox.tsx

```diff
@@ -0,0 +1,31 @@
+import * as S from './AllCheckBox.css';
+
+import { User } from 'apis/getFollowData';
+
+interface AllCheckBoxProps {
+  isAllChecked: boolean;
+  setIsAllChecked: React.Dispatch<React.SetStateAction<boolean>>;
+  list: User[];
+  setCheckList: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+export default function AllCheckbox({ isAllChecked, setIsAllChecked, list, setCheckList }: AllCheckBoxProps) {
+  const handleOnChange = () => {
+    setIsAllChecked(!isAllChecked);
+    setCheckList([]);
+
+    if (!isAllChecked) {
+      for (const user of list) {
+        setCheckList((prev) => [...prev, user.login]);
+      }
+    }
```

Comment:
다음과 같이 간결하게 바꿔볼 수 있을 것 같아요.

```suggestion
    if (isAllChecked) {
        setCheckList([]);
    } else {
        setChecklist(list.map(user => user.login));
    }
```